backend/api/firebase_auth.py

```
@@ -0,0 +1,64 @@
+                user = User(
+                    username=uid,
+                    email=email,
+                    is_active=True
+                )
+                user.save()
+            
+            # Return the user and token
+            return (user, decoded_token)
+        except auth.InvalidIdTokenError:
+            raise exceptions.AuthenticationFailed('Invalid token')
+        except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)